### PR TITLE
[Google Cloud PubSub] Fix getting token when running in emulated mode

### PR DESCRIPTION
**Pull request by damianbl**
- it should not try to fetch the token from the session when running in emulated mode
- fixes [981](https://github.com/akka/alpakka/issues/981)

### CODE REVIEW

File: google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala [hunk 1]
```diff
@@ -31,13 +31,27 @@ private final class GooglePubSubSource(projectId: String,
...
 
-        val req = session.getToken().flatMap { token =>
-          httpApi.pull(project = projectId, subscription = subscription, accessToken = token, apiKey = apiKey)
-        }
-        req.onComplete { tr =>
-          callback.invoke(tr -> mat)
+        if (httpApi.isEmulated) {
+          httpApi
+            .pull(project = projectId, subscription = subscription, maybeAccessToken = None, apiKey = apiKey)
+            .onComplete { tr =>
```
Comment (ennru) [inline google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala]: To make it obvious that the `onComplete` is shared between both branches you might go back to the `val req` by assigning the if clause to it.

Comment (damianbl) [reply google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala]: Sure, changed. Is this what you meant?

File: google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala [hunk 2]
```diff
@@ -31,13 +31,27 @@ private final class GooglePubSubSource(projectId: String,
       def fetch(implicit mat: Materializer): Unit = {
         import mat.executionContext
 
-        val req = session.getToken().flatMap { token =>
-          httpApi.pull(project = projectId, subscription = subscription, accessToken = token, apiKey = apiKey)
+        val req = if (httpApi.isEmulated) {
+          val resp = httpApi
+            .pull(project = projectId, subscription = subscription, maybeAccessToken = None, apiKey = apiKey)
+          state = Fetching
```
Comment (ennru) [inline google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala]: If you move the state change to its original spot it get even better...

Comment (damianbl) [reply google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala]: Fixed.

**Comment by ennru** [comment]
Thank you for fixing this!

**Comment by damianbl** [comment]
Thanks for accepting this. 
When we have it released, it will resolve the last issue I have when integrating with Google PubSub using Alpakka.

Thanks!

**Pull request merged**